src/Services/RabbitMQService.cs

```diff
@@ -10,15 +10,50 @@ internal sealed class RabbitMQService : IRabbitMQService
     {
         private IModel _channel;
         private IBasicPublishBatch _batch;
+        private string _connectionString;
         private string _hostName;
         private string _queueName;
+        private string _userName;
+        private string _password;
+        private int _port;
 
-        public RabbitMQService(string hostName, string queueName)
+        public RabbitMQService(string connectionString, string hostName, string queueName, string userName, string password, int port)
         {
-            _hostName = hostName ?? throw new ArgumentNullException(nameof(hostName));
+            _connectionString = connectionString;
+            _hostName = hostName;
             _queueName = queueName ?? throw new ArgumentNullException(nameof(queueName));
+            _userName = userName;
+            _password = password;
+            _port = port;
+
+            ConnectionFactory connectionFactory = new ConnectionFactory();

```

Comment:
Move logic that builds connectionFactory to a separate method. Add unit tests that verify various if conditions